### PR TITLE
feat: upgrade to min deployment version to iOS 12

### PR DESCRIPTION
**Pull request by lawmicha**
*Issue #, if available:*

*Description of changes:*
- This PR updates the iOS deployment target version to iOS 12, to anticipate a dependency upgrade from AppSyncRealTimeClient. https://github.com/aws-amplify/aws-appsync-realtime-client-ios/pull/156

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by 5d — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ Carthage
 Pods
 .build/
 build/
+Pods/
```
Comment (ruisebas) [inline .gitignore:10]: Isn't this included with the `Pods` in line 7? 🤷‍♂️

File: AWSAppSyncClient.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -2387,7 +2387,7 @@
 				CODE_SIGN_STYLE = Automatic;
 				GCC_C_LANGUAGE_STANDARD = gnu11;
 				INFOPLIST_FILE = AWSAppSyncTestAppUITests/Info.plist;
-				IPHONEOS_DEPLOYMENT_TARGET = 12.1;
```
Comment (ruisebas) [inline AWSAppSyncClient.xcodeproj/project.pbxproj:2390]: You've downgraded some targets from 12.**1** to 12.**0**. 

Just to be sure, did you validate that these changes are safe?

Comment (5d) [reply AWSAppSyncClient.xcodeproj/project.pbxproj:2390]: Here the `AWSAppSyncTestAppUITests` is an empty test project. I've verified all the other downgraded targets are success to build.

File: .gitignore (hunk 1 above)
Comment (5d) [reply .gitignore:10]: oh, this is due to a auto merge with main. I will remove this line

**Review by ruisebas — APPROVED** [review]

**Pull request merged**